src/routes/genres.routes.ts

```diff
@@ -0,0 +1,182 @@
+import { Router, Request, Response, NextFunction } from 'express';
+
+import { GenreModel } from '../models/genres.schemas';
+import { Genre } from '../interfaces/interfaces';
+import validateGenre from '../middleware/genres.validation';
+
+const router: Router = Router();
+
+/**
+ * @swagger
+ *    tags:
+ *    name: Genres
+ * /genres:
+ *   get:
+ *     summary: Get genres list
+ *     tags:
+ *       - Genres
+ *     responses:
+ *       200:
+ *         description: List of genres
+ *         content:
+ *           application/json:
+ *             example:
+ *               - name: Action
+ *               - name: Drama
+ *       404:
+ *         description: Genres not found
+ *       500:
+ *         description: Internal Server Error
+ */
+router.get('/', async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const genres: Genre[] = await GenreModel.find();
+
+    if (genres.length === 0) {
+      return res.status(200).json({ data: 'Genres not added yet' });
+    }
+
+    return res.status(200).json({ data: genres });
+  } catch (error) {
+    next(error);
+  }
+});
+
+/**
+ * @swagger
+ * /genres:
+ *   post:
+ *     summary: Create new genre
+ *     tags:
+ *       - Genres
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           example:
+ *             name: New genre
+ *     responses:
+ *       201:
+ *         description: Genre created successfully
+ *         content:
+ *           application/json:
+ *             example:
+ *               name: New genre
+ *       400:
+ *         description: Bad Request
+ *       500:
+ *         description: Internal Server Error
+ */
+router.post(
+  '/',
+  validateGenre,
```

Comment:
For improvements, you can  invoke validateGenre(genreSchema or movieSchema) - to not dublicate the same validate functions twice in middlewares folder